trajectory_executor/src/trajectory_executor/trajectory_executor.cpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2018-2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include "trajectory_executor/trajectory_executor.hpp"
+#include <ros/ros.h>
+#include <utility>
+#include <cav_msgs/SystemAlert.h>
+
+namespace trajectory_executor 
+{
+    cav_msgs::TrajectoryPlan trimFirstPoint(const cav_msgs::TrajectoryPlan plan) {
+        cav_msgs::TrajectoryPlan out(plan);
+        out.trajectory_points.erase(out.trajectory_points.begin());
+        return out;
+    }
+
+    TrajectoryExecutor::TrajectoryExecutor(int traj_frequency) :
+        _min_traj_publish_tickrate_hz(traj_frequency),
+        _timesteps_since_last_traj(0)
+    {
+    }
+
+    TrajectoryExecutor::TrajectoryExecutor() :
+        _min_traj_publish_tickrate_hz(10),
+        _timesteps_since_last_traj(0)
+    {
+    }
+
+    std::map<std::string, std::string> TrajectoryExecutor::queryControlPlugins()
+    {
+        // Hard coded stub for MVP since plugin manager won't be developed yet
+        // TODO: Query plugin manager to receive actual list of plugins and their corresponding topics
+        ROS_DEBUG("Executing stub behavior for plugin discovery MVP...");
+        std::map<std::string, std::string> out;
+
+        std::string default_control_plugin;
+        _default_nh->param<std::string>("default_control_plugin", default_control_plugin, "NULL");
+
+        std::string default_control_plugin_topic;
+        _default_nh->param<std::string>("default_control_plugin_topic", default_control_plugin_topic, "NULL");
+
+        out[default_control_plugin] = default_control_plugin_topic;
+        return out;
+    }
+    
+    void TrajectoryExecutor::onNewTrajectoryPlan(cav_msgs::TrajectoryPlan msg)
```

Comment:
What is the reason for passing by value in this function?

Reply:
Was just the easiest way to get an object i could pass into a copy constructor, wasn't having much luck with their `cav_msgs::TrajectoryPlan::ConstPtr&` interface here because getting the actual object out of the `ConstPtr&` (I think under the hood it's a `boost::shared_ptr`) was being a real pain.

Reply:
Yep. I was using ```boost::shared_ptr``` but I do not think it is a big deal for this small message.